### DISCUSSION TITLE
Error in Pulsar-broker during Milvus-Helm deployment

### DISCUSSION THREAD

**Discussion by wangchao6856** [milvus-io/milvus#37358] [Q&A and General discussion]
Using milvus-helm to deploy Milvus, the InitContainers section of the pulsar-broker reports an error indicating a failure to connect to Zookeeper. The specific command is:
```
until bin/bookkeeper org.apache.zookeeper.ZooKeeperMain -server milvus-ka0sje-pulsar-zookeeper:2181 get /admin/clusters/milvus-ka0sje-pulsar; do
echo "pulsar cluster milvus-ka0sje-pulsar isn't initialized yet ... check in 3 seconds ..." && sleep 3;
done;
```
The Zookeeper node has started successfully, and I used a Python script to connect to Zookeeper on port 2181, which shows that the root directory contains the zookeeper and ledgers directories, but there is no admin directory. So, how is the /admin directory created?

![image](https://github.com/user-attachments/assets/f5411b41-51fc-4f5c-a9d5-11c3c9b50dca)

![image](https://github.com/user-attachments/assets/ecc651ec-bdfa-4e23-b55a-c70e99bb8895)

Although the Zookeeper node has started successfully, there are errors in the logs

![image](https://github.com/user-attachments/assets/5bb9021d-341b-46d0-8c77-bb784fd82fc9)

**Comment by xiaofan-luan** [comment]
@wangchao6856 
seems that it's the pulsar etcd requires the resource. Did you by chance fix that already?

**Comment by haorenfsa** [comment]
@wangchao6856 
> So, how is the /admin directory created?

It's created through an init `job` created by the helm chart, called `${releaseName}-pulsar-init`. It does the pulsar metadata initialization. It's followed by another init `job` called `${releaseName}-bookie-init` to initialize meta for the bookie of pulsar.
So it's usually normal for `the InitContainers of the pulsar-broker` to report such error while these jobs are still doing.
But if you can't find those jobs in your Kubernetes. Try uninstall and install the helm chart again.
